components/eamxx/src/physics/shoc/impl/shoc_main_impl.hpp

```diff
@@ -310,6 +311,12 @@ void Functions<S,D>::shoc_main_internal(
           workspace,                      // Workspace
           pblh);                          // Output
 
+  // Assign tkh to the tkh output variable
+  const Int nlev_pack = ekat::npack<Spack>(nlev);
+  Kokkos::parallel_for(Kokkos::TeamVectorRange(team, nlev_pack), [&] (const Int& k) {
+    tkh_out(k) = tkh(k);
```

Comment:
If you have tkh as inout arg (b/c presumably you need it outside), then there's no point in using a slot from the workspace manager for `tkh`. We could remove that slot, and rename the inout arg `tkh`...

@tcclevenger may also know whether removing this slot would entail a reduction in the overall WSM needs for shoc or not...

Reply:
Yeah, I believe this reduces WSM shoc needs since `tkh` is global to `shoc_main()`, which means when we get rid of it we will need to reduce the WSM request size for the internal buffer in `eamxx_shoc_process_interface`. 

Also for small kernels I'm guessing we can remove the `tkh` temporary.